File: backend/src/main/java/org/law_app/backend/entity/Company.java
package org.law_app.backend.entity;

import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.util.List;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@EntityListeners(AuditingEntityListener.class)
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Company {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    String id;

    String name;             // Tên công ty
    String representative;   // Người đại diện
    String taxCode;          // Mã số thuế
    String websiteName;      // Tên website
    String email;            // Email liên hệ

    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
    @JoinColumn(name = "company_id")
    List<Location> locations;

    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
    @JoinColumn(name = "company_id")
    List<Important> importants;

    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
    @JoinColumn(name = "company_id")
    List<PhoneContact> phoneContacts;

    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
    @JoinColumn(name = "company_id")
    List<Social> socials;
}

File: frontend/src/types/company.ts
export interface Company {
    id: string;
    name: string;
    representative: string;
    taxCode: string;
    websiteName: string;
    email: string;
  }
  export interface Location {
    id: string;
    type: string;
    address: string;
    color: string;
  }
  export interface PhoneContact {
    id: string;
    label: string;
    number: string;
    color: string;
  }
  export interface Important {
    id: string;
    icon: string; // icon name, e.g., "Shield", "FileText"
    text: string;
    href: string;
    color: string;
  }
  export interface Social {
    id: string;
    icon: string; // icon name, e.g., "Facebook", "Linkedin"
    href: string;
    label: string;
    color: string;
  }
  export interface TotoCompany{
    company: Company;
    locations: Location[];
    phoneContacts: PhoneContact[];
    importants: Important[];
    socials: Social[];
  }